keps/sig-auth/20190607-certificates-api.md

```diff
@@ -132,6 +133,31 @@ A typical successful issuance proceeds as follows.
    the `.Status.Certificate` field.
 1. The requestor observes the update, and stores the certificate locally.
 
+### Signers
+CSRs have a `signerName` field which is used to specify which signer the CSR creator wants to sign the certificate.
```

Comment:
It would be good to add a general description of what a definer of a given signerName is responsible for specifying:
* permitted subjects (any? specific subtree?) and behavior when a disallowed subject is requested
* permitted x509 extensions (IP SANs? DNS SANs? Email SANs? URI SANs? others?) and behavior when a disallowed extension is requested
* expiration/cert lifetime (fixed by signer? configurable by admin? CSR-determined?) and behavior when an expiration different than the signer-determined expiration is specified in the CSR
* permitted key usages / extended key usages (client only? server only? any? signer-determined? CSR-determined?) and behavior when usages different than the signer-determined usages are specified in the CSR
* CA bit allowed/disallowed, and behavior if a CSR contains a request a for a CA cert when the signer does not permit it

Then define each of those items for the four `kubernetes.io/...` signerNames being proposed. Since deployers can replace the KCM-based signer with their own, we need to be very clear about what the expected behavior of that signer is for the `kubernetes.io/...` certs.

Reply:
done